yt/frontends/ramses/data_structures.py

```diff
@@ -109,37 +105,62 @@ def check_standard_files(folder, iout):
         ok &= (folder / f"info_{iout}.txt").is_file()
         return ok
 
+    @staticmethod
+    def _match_output_and_group(
+        path: Path,
+    ) -> Tuple[Path, Optional[Path], Optional[str]]:
+
+        # Three cases for path
+        # 1. path == output_XXXXX and group_00001 exists
+        # 2. path == output_XXXXX and group_00001 does not exist
+        # 3. path == output_XXXXX/group_YYYYYY
+        iout_match = OUTPUT_DIR_RE.match(path.name)
+        if iout_match:
+            group_dir = path / "group_00001"
+
+            return path, group_dir if group_dir.is_dir() else None, iout_match.group(1)
+        else:
+            iout_match = OUTPUT_DIR_RE.match(path.parent.name)
+
+            return path.parent, path, iout_match.group(1) if iout_match else None
```

Comment:
honestly this is still hardly readable to me; to the point it's not easy to find a better alternative.*
Can we having ternaries inside tuples maybe ?

Reply:
I'm also struggling to convince myself we actually need this double match now. Can't we just get everything we need by trying to match the regex to the parent dir in all cases ?